File: web/plugins/lcm.js
import { WebPlugin } from "../engine/plugin.js";

export class LcmPlugin extends WebPlugin {
    constructor() {
        super("最小公倍数");
    }

    async execute(line, context) {
        const match = line.match(/^最小公倍数\s+(.+)\s+(.+)$/);
        if (!match) return;

        let a = Math.abs(Number(
            context.parser.parseValue(match[1], context.runtime)
        ));

        let b = Math.abs(Number(
            context.parser.parseValue(match[2], context.runtime)
        ));

        const originalA = a;
        const originalB = b;

        while (b !== 0) {
            [a, b] = [b, a % b];
        }

        context.runtime.write(
            originalA === 0 || originalB === 0
                ? 0
                : Math.abs(originalA * originalB) / a
        );
    }
}

File: pc/plugins/lcm.py
import re
from core.plugin import Plugin


class LcmPlugin(Plugin):
    def translate(self, line):
        match = re.fullmatch(r'最小公倍数\s+(.+)\s+(.+)', line)
        if not match:
            return None

        a = match.group(1)
        b = match.group(2)

        return f"print(__import__('math').lcm({a}, {b}))"